contracts/service-registry/src/contract.rs

```diff
@@ -255,9 +255,7 @@ pub mod execute {
             .may_load(deps.storage, (&service_name, &info.sender))?
             .ok_or(ContractError::WorkerNotFound)?;
 
-        for chain in chains {
-            WORKERS_PER_CHAIN.save(deps.storage, (&service_name, &chain, &info.sender), &())?;
-        }
+        let _res = register_chain_support(deps.storage, service_name.clone(), chains, info.sender);
```

Comment:
my bad, I missed the "s" in `chains`. I'd keep the names identival, i.e. make the call to `state::register_chains_support`

Reply:
Changed! Thanks.